src/main/java/uk/gov/hmcts/reform/hmc/config/MessageProcessor.java

```diff
@@ -6,28 +6,45 @@
 import uk.gov.hmcts.reform.hmc.ApplicationParams;
 import uk.gov.hmcts.reform.hmc.client.futurehearing.ActiveDirectoryApiClient;
 import uk.gov.hmcts.reform.hmc.client.futurehearing.HearingManagementInterfaceApiClient;
+import uk.gov.hmcts.reform.hmc.errorhandling.MalformedMessageException;
 import uk.gov.hmcts.reform.hmc.repository.DefaultFutureHearingRepository;
 
 import java.util.Map;
 
+import static uk.gov.hmcts.reform.hmc.config.MessageType.AMEND_HEARING;
+import static uk.gov.hmcts.reform.hmc.config.MessageType.DELETE_HEARING;
+
 @Slf4j
 @Component
 public class MessageProcessor {
 
     private final ApplicationParams applicationParams;
     private final ActiveDirectoryApiClient activeDirectoryApiClient;
     private final HearingManagementInterfaceApiClient hmiClient;
+    private static final String MESSAGE_TYPE = "message_type";
+    private String caseListingID;
```

Comment:
This should be a local var (e.g. just before L56) instead of at the class level to avoid issues - similar to how `MessageType` is initialised (with an initially unknown value) at L41